tiny_dnn/core/framework/tensor.h

```diff
@@ -231,150 +188,90 @@ class Tensor {
-   * with offset zero
-   *
+   * Reshape tensor
+   * @param sz
    */
-  Tensor subView(std::initializer_list<size_t> const &new_shape) {
-    return subview_impl({}, new_shape);
-  }
+  void reshape(const std::vector<size_t> &sz) { storage_.reshape(sz); }
 
-  /**
-   * @brief Returns a sub view from the current tensor with a given size.
-   * The new tensor will share data with its parent tensor so that each time
-   * that data is modified, it will be updated in both directions.
-   *
-   * The new sub view tensor will be extracted assuming continuous data.
-   *
-   * @param start The offset from the parent tensor
-   * @param new_shape The size for the new tensor
-   * @return An instance to the new tensor
-   *
-   * Usage:
-   *
-   *  Tensor<float_t, 4> t({2,2,2,2});                   // we create a 4D
-   * tensor
-   *  Tensor<float_t, 4> t_view = t.view({2,2}, {2,2});  // we create a 2x2
-   * matrix view from
-   *                                                     // offset 4.
-   */
-  Tensor subView(std::initializer_list<size_t> const &start,
-                 std::initializer_list<size_t> const &new_shape) {
-    return subview_impl(start, new_shape);
-  }
+  void resize(const std::vector<size_t> &sz) { storage_.reshape(sz); }
+
+  // size_t size() const { return size_; }
+
+  Tensor operator[](size_t index) { return Tensor(storage_[index]); }
 
   /**
-   * @brief Returns whether the tensor is a view of another tensor
+   * Returns new Tensor which has shared storage with current, but different
+   * shape
    *
+   * Example:
+   * @code
+   * Tensor b({4,2});
+   * a = b.subView({2,2,2};
+   * @endcode
+   * b is Tensor of shape 4x2 and a is Tensor of shape 2x2x2. Changing a(0,0,0)
+   * will change b(0,0) too.
+   * @param new_shape
+   * @return
    */
-  bool isSubView() const { return size_ != storage_ptr_->size(); }
+  /*
+ Tensor<U,xt::xbroadcast<Storage, std::vector<size_t>>>
+ subView(std::initializer_list<size_t> const &new_shape) {
+   auto res = Tensor<U,xt::xbroadcast<Storage,
+ std::vector<size_t>>>(xt::broadcast(storage_, storage_.shape()));
+   res.storage_.reshape(new_shape);
+   return res;
+ }*/
 
- private:
   /**
-   * Constructor that accepts a pointer to existing TensorStorage, together
-   * with shape and offset.
-   * @param storage pointer to TensorStorage
-   * @param offset offset from first element of storage
-   * @param shape shape of the Tensor
-   * @return
+   * Returns view of current Tensor
+   * @tparam Values index type
    */
-  explicit Tensor(const TensorStoragePointer storage,
-                  const size_t offset,
-                  std::initializer_list<size_t> const &shape) {
-    offset_      = offset;
-    size_        = product(shape);
-    storage_ptr_ = storage;
-    std::copy(shape.begin(), shape.end(), shape_.begin());
+  template <typename... Values>
+  Tensor<U, xt::xview<Storage &, xt::xrange<Values>...>> subView(
+    std::initializer_list<Values>... lists) {
```

Comment:
I don't think it can be any container, as the size has to be known at compile time.

If you want "dynamic views" you can have a look at the `dynamic_view` implementation in `xstridedview.hpp` (might be useful for the future). Usage is not really well documented right now, but there are some examples in `test_dynamicview.cpp`.